### PR TITLE
feat: create new products

### PR DESCRIPTION
**Pull request by v-racine**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -90,6 +98,31 @@ const AppFactory = (args) => {
     res.send(productsNewTemplate({}));
   });
 
+  app.post('/admin/products/new', [parseTitle, parsePrice], async (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.send(productsNewTemplate({ errors }));
+    }
+    //
```
Comment (w-h-a) [inline src/app.js:107]: why, monsieur?

File: tests/newProduct.test.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+const AppFactory = require('../src/app');
+const request = require('supertest');
+
+describe('create new product', () => {
+  const mockProductsRepo = {};
+
+  const app = AppFactory({
+    productsRepo: mockProductsRepo,
+  });
+
```
Comment (w-h-a) [inline tests/newProduct.test.js:14]: HELLLLLL YEAH

**Review by w-h-a — APPROVED** [review]
🐑🇮🇹

File: tests/newProduct.test.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    });
+
+    test("then: we return 'Submitted!'", async () => {
+      const status = rsp.status;
+      expect(status).toBe(200);
+
+      const text = rsp.text;
+      expect(text).toMatchSnapshot();
+
+      expect(mockProductsRepo.getOneBy).toHaveBeenCalledWith({ title: 'testing' });
```
Comment (w-h-a) [inline tests/newProduct.test.js:140]: can we also assert on what we sent to `.create`?

**Review by w-h-a — CHANGES_REQUESTED** [review]
one test change

File: src/app.js (hunk 1 above)
Comment (v-racine) [reply src/app.js:107]: What do you mean 'why?', M?

**Review by w-h-a — APPROVED** [review]

**Pull request merged**